consul watch config panic

<!--
Please answer these questions before submitting your issue. Thanks!
For questions please use one of our forums: https://go-kratos.dev/docs/getting-started/faq
-->
#### What happened:
```go
if err := c.Watch("service.name", func(key string, value config.Value) {
  fmt.Printf("config changed: %s = %v\n", key, value)
  // 在这里写回调的逻辑
}); err != nil {
  log.Error(err)
}
```
#### What you expected to happen:
log err
#### How to reproduce it (as minimally and precisely as possible):
modify a config value string to int, then program panic exit
#### Anything else we need to know?:

#### Environment:
- Kratos version (use `kratos -v`): v2.1.5
- Go version (use `go version`): go 1.17.6
- OS (e.g: `cat /etc/os-release`): centos7
- Others:

> please show me your panic log.

···go
panic: sync/atomic: store of inconsistently typed value into Value

goroutine 38 [running]:
sync/atomic.(*Value).Store(0xc000518590, {0xc61760, 0x143f938})
        /usr/local/go/src/sync/atomic/value.go:77 +0xd8
github.com/go-kratos/kratos/v2/config.(*config).watch.func1({0xc62060, 0xc0005185a0}, {0xd5dc40, 0xc000518590})
        /root/go/pkg/mod/github.com/go-kratos/kratos/v2@v2.1.5/config/config.go:90 +0x130
sync.(*Map).Range(0xc000410190, 0xc0000f5f98)
        /usr/local/go/src/sync/map.go:346 +0x2aa
github.com/go-kratos/kratos/v2/config.(*config).watch(0xc0004329c0, {0xeea5e0, 0xc000408900})
        /root/go/pkg/mod/github.com/go-kratos/kratos/v2@v2.1.5/config/config.go:86 +0x205
created by github.com/go-kratos/kratos/v2/config.(*config).Load
        /root/go/pkg/mod/github.com/go-kratos/kratos/v2@v2.1.5/config/config.go:119 +0x2c5

···

--- a/config/config.go
+++ b/config/config.go
@@ -86,7 +86,7 @@ func (c *config) watch(w Watcher) {
 		c.cached.Range(func(key, value interface{}) bool {
 			k := key.(string)
 			v := value.(Value)
-			if n, ok := c.reader.Value(k); ok && !reflect.DeepEqual(n.Load(), v.Load()) {
+			if n, ok := c.reader.Value(k); ok && reflect.TypeOf(n.Load()) == reflect.TypeOf(v.Load()) && !reflect.DeepEqual(n.Load(), v.Load()) {
 				v.Store(n.Load())
 				if o, ok := c.observers.Load(k); ok {
 					o.(Observer)(k, v)

